src/traits/abi.rs

```diff
@@ -0,0 +1,57 @@
+use crate::*;
+
+/// Provides a generic way of referring to and converting between a Rust object
+/// and its WinRT ABI equivalent.
+///
+/// This trait is automatically used by the generated bindings and should not be
+/// used directly.
+pub unsafe trait Abi: Sized {
+    type Abi;
+
+    // abi must always transmute (blit) and never Copy.
+    unsafe fn abi(&self) -> Self::Abi {
```

Comment:
Why is this method `unsafe`?

Reply:
The call is not unsafe in itself but anything you do with the resulting value is almost assuredly unsafe. Not sure where that leaves this method. In C++/WinRT calling the equivalent is always a sign of danger.

Reply:
In Rust the convention is that functions that are safe to call (even if they give you a value that is impossible to use without using `unsafe` are still considered safe. For example, [`str::as_mut_ptr`](https://doc.rust-lang.org/stable/std/primitive.str.html#method.as_mut_ptr) is safe even though the pointer it returns is very difficult to use correctly and should only ever be used in extreme situations.